app/src/main/java/protect/card_locker/LoyaltyCardViewActivity.java

```diff
@@ -634,7 +635,15 @@ public void onResume() {
         int darkenedColor = ColorUtils.blendARGB(backgroundHeaderColor, Color.BLACK, 0.1f);
         binding.barcodeScaler.setProgressTintList(ColorStateList.valueOf(darkenedColor));
         binding.barcodeScaler.setThumbTintList(ColorStateList.valueOf(darkenedColor));
+
+        // Set bottomAppBar and system navigation bar color
         binding.bottomAppBar.setBackgroundColor(darkenedColor);
+        if (Build.VERSION.SDK_INT >= 27) {
```

Comment:
I'm slightly confused here because at the start of this function we have this:

```java
        Window window = getWindow();
        if (window != null) {
```

Is there a null check missing here? Or is that not needed for some reason? Also can we reuse the `window` variable or do we need 3 calls to `getWindow()`?

Reply:
Missed from my side, should be reused and have a null check.